### PR TITLE
Trapping rain water in python

### LINKED ISSUE #5851: Trapping rain water in python
## 💥 Propose

Add code for trapping rain water in python

### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#reporting-new-issues)?

Yes

### PR DESCRIPTION
**Pull request by Amit366**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

Added code for trapping rain water

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Fixes: #5851 

Replit: https://replit.com/@AmitKumarKuma29/IllTepidAddons#main.py

### CODE REVIEW

**Comment by Amit366** [comment]
@HarshCasper please review

**Comment by Amit366** [comment]
@ankitaggarwal23 please review

File: Python/cp/Trapping_rain_water.py [hunk 1]
```diff
@@ -0,0 +1,65 @@
+# Given n non-negative integers representing an elevation map where the 
+# width of each bar is 1, compute how much water it can trap after raining.
```
Comment (ankitaggarwal23) [inline Python/cp/Trapping_rain_water.py]: use Multi-Line Comments

File: Python/cp/Trapping_rain_water.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+# Example 2:
+
+# Input:
+# Enter number of elements: 
+# 5
+# Enter the elements: 
+# 1 4 2 0 1
+
+# Output:
+# Water it can trap is: 1
```
Comment (ankitaggarwal23) [inline Python/cp/Trapping_rain_water.py]: remove extra test case

File: Python/cp/Trapping_rain_water.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+# Input:
+# Enter number of elements: 
+# 12
+# Enter the elements: 
+# 0 1 0 2 1 0 1 3 2 1 2 1
+
+# Output:
+# Water it can trap is: 6
+
```
Comment (ankitaggarwal23) [inline Python/cp/Trapping_rain_water.py]: use  Multi-Line Comments

**Review by ankitaggarwal23 — CHANGES_REQUESTED** [review]

File: Python/cp/Trapping_rain_water.py (hunk 2 above)
Comment (Amit366) [reply Python/cp/Trapping_rain_water.py]: @ankitaggarwal23 so only one test case??

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**